tests/fused_attention_stablehlo_test.py

```diff
@@ -543,5 +574,170 @@ def test_sdpa_utils(self):
             query, key, AttentionLayout.BNTH.value, cudnn_version, has_bias,
             is_training)
 
+
+@jtu.with_config(jax_numpy_dtype_promotion='standard')
+class DotProductAttentionF8Test(jtu.JaxTestCase):
+  def setUp(self):
+    super().setUp()
+    if jax.device_count() < 4:
+      self.skipTest("Requires more than 4 devices.")
+    try:
+      cudnn_version = check_cudnn_version()
+    except RuntimeError as e:
+      self.skipTest(str(e))
+      return
+    if cudnn_version < 9010:
+      self.skipTest("Requires >= cuDNN 9.1.0")
+    if not jtu.is_cuda_compute_capability_at_least("9.0"):
+      self.skipTest("Requires at least Hopper arch")
+
+  @jtu.sample_product(
+      batch_size=[2, 4],
+      seq_len=[128, 256],
+      num_heads=[4, 8],
+      head_dim=[128],
+      mask_type=[MaskType.NO_MASK],
+      scale=[1.0, 0.75],
+      dtype=[jnp.bfloat16, jnp.float16]
+  )
+  @jtu.run_on_devices("cuda")
+  def test_sdpa_fp8(self, batch_size: int, seq_len: int, num_heads: int,
+                    head_dim: int, mask_type: MaskType,
+                    scale: float, dtype: jnp.dtype):
+    k1, k2, k3, k4 = jax.random.split(jax.random.key(0), 4)
+    input_shape = (batch_size, seq_len, num_heads, head_dim)  # only test the default BTNH
+    query_h = jax.random.normal(
+        k1, input_shape, dtype=dtype)
+    key_h = jax.random.normal(
+        k2, input_shape, dtype=dtype)
+    value_h = jax.random.normal(
+        k3, input_shape, dtype=dtype)
+    grad_h = jax.random.normal(
+        k4, input_shape, dtype=dtype)
+    query = cast_to_representable(query_h, jnp.float8_e4m3fn)
+    key = cast_to_representable(key_h, jnp.float8_e4m3fn)
+    value = cast_to_representable(value_h, jnp.float8_e4m3fn)
+    grad = cast_to_representable(grad_h, jnp.float8_e4m3fn)
+
+    query_quantized = quantize(query, jnp.float8_e4m3fn, jnp.ones((1,)), jnp.float32)
+    key_quantized = quantize(key, jnp.float8_e4m3fn, jnp.ones((1,)), jnp.float32)
+    value_quantized = quantize(value, jnp.float8_e4m3fn, jnp.ones((1,)), jnp.float32)
+    grad_quantized = quantize(grad, jnp.float8_e4m3fn, jnp.ones((1,)), jnp.float32)
+
+    devices = np.array(jax.local_devices()[:4])
+    devices = devices.reshape((2, 2))
+
+    with Mesh(devices, ("dp", "tp")) as mesh:
```

Comment:
Please don't use the context manager.

Reply:
May I know about the background since the original tests use them.

Reply:
original tests shouldn't have and mesh does not really work with jit. So it's better to remove it and in the other tests too.

Reply:
IIUC from Yash, we expect things to Just Work after deleting this `with Mesh(...)` context manager.

Reply:
IIUC from Yash, we expect things to Just Work after deleting this `with Mesh(...)` context manager.